src/store/quiz.js

```diff
@@ -0,0 +1,89 @@
+import QuizService from '@/api-services/quiz.service'
+
+const state = {
+  newQuiz: {
+    name: 'Nowy quiz',
+    description: 'Opis quizu',
+    image: {},
+    questions: []
+  }
+}
+
+const getters = {
+  newQuiz () {
+    return state.newQuiz
+  },
+  newQuestion () {
+    return (questionIndex) => {
+      return state.newQuiz.questions[questionIndex]
+    }
+  },
+  newChoice () {
+    return (questionIndex, choiceIndex) => {
+      return state.newQuiz.questions[questionIndex].choices[choiceIndex]
+    }
+  }
+}
+
+const actions = {
+  create ({ commit }, quizData) {
+    return QuizService.create(quizData).then((data) => {
+    /**
+    * @todo inform user about result of this operation and set state if needed.
+    */
+    })
+  }
+}
+
+const mutations = {
```

Comment:
One more additional thing. For readability purposes I use uppercase mutation name f.e. ADD_CHOICE() with following action addChoice(). It's not necessary, but I prefer that style. Good job with refactor!

Reply:
I changed to this convention and used file with declaration of mutation names to make changes to them faster.